grails-app/services/org/pih/warehouse/inventory/CycleCountService.groovy

```diff
@@ -421,12 +421,34 @@ class CycleCountService {
 
     private void determineCycleCountItemStatusForSubmit(CycleCountItem cycleCountItem, boolean requireRecountOnDiscrepancy) {
         if ((cycleCountItem.quantityOnHand == cycleCountItem.quantityCounted) || !requireRecountOnDiscrepancy) {
-            cycleCountItem.status = CycleCountItemStatus.READY_TO_REVIEW
+            // TODO: Once we add support for the "to review" tab, this should be changed to assign status =
+            //       CycleCountItemStatus.READY_TO_REVIEW. For now we simply complete the count for the item.
+            cycleCountItem.status = CycleCountItemStatus.APPROVED
             return
         }
         cycleCountItem.status = CycleCountItemStatus.COUNTED
     }
 
+    /**
+     * Given a resolved cycle count, close it out by bringing it to the completion state and committing any
+     * required quantity adjustments.
+     */
+    private void closeCycleCount(CycleCount cycleCount, boolean refreshQuantityOnHand) {
+        // If the count was cancelled, there's nothing to do except also cancel the request
+        if (cycleCount.status == CycleCountStatus.CANCELED) {
+            cycleCount.cycleCountRequest.status = CycleCountRequestStatus.CANCELED
+            return
+        }
+
+        if (cycleCount.status != CycleCountStatus.COMPLETED) {
+            throw new IllegalStateException("Cannot complete cycle count when it's in status ${cycleCount.status}")
+        }
+
+        // The count completed successfully so update the request status to match and commit the adjustments.
+        cycleCount.cycleCountRequest.status = CycleCountRequestStatus.COMPLETED
+        cycleCountTransactionService.createTransactions(cycleCount, refreshQuantityOnHand)
```

Comment:
Would it not be better to create a transaction and close afterward?

Reply:
sure I can flip these but this whole call is wrapped in a `@Transactional` so ultimately the order shouldn't matter much. If it fails at any point the whole thing should get rolled back